### PR TITLE
UserData class

### PR DESCRIPTION
**Pull request by SmithAF**
Tasks:
- [x] Documentation
- [x] Unit Tests
- [x] Typescript Typings
- [ ] Haxe Externs
- [x] `npm run build:full`

Added UserData class to handle data communication between SpringRoll Connect and SpringRoll

### CODE REVIEW

File: src/state/UserData.spec.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { UserData } from './UserData';
+
+describe('UserData', () => {
+  it('should construct', () => {
+    new UserData();
```
Comment (chipbell4) [inline src/state/UserData.spec.js]: Could we make this test a little more robust? Perhaps mock `comm` somehow?

File: src/state/UserData.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * Sends data to SpringRoll Container
+   * @memberof UserData
+   * @param {*} value
+   * @param {string} name
+   * @static
+   */
+  static write(name, value) {
+    const warning = `Could not complete write action for ${name} with value ${value}. Bellhop is not connected.`;
+    comm.connected ? comm.send(WRITE, { name, value }) : console.warn(warning);
```
Comment (chipbell4) [inline src/state/UserData.js]: Could we return a promise here?

File: src/state/UserData.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * Gets data from SpringRoll Container
+   * @memberof UserData
+   * @param {function} callback
+   * @param {string} name
+   * @static
+   */
+  static read(name, callback) {
+    const warning = `Could not complete read action for ${name}. Bellhop is not connected.`;
+    comm.connected
```
Comment (chipbell4) [inline src/state/UserData.js]: Could we return a promise here?

File: src/communication/comm.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { Bellhop } from 'bellhop-iframe';
+
+export const comm = new Bellhop();
```
Comment (chipbell4) [inline src/communication/comm.js]: Could we rename this? I don't think `comm` clearly indicates intention. Something like `bellhopSingleton`. It's longer but clearer.

Comment (SmithAF) [reply src/communication/comm.js]: Yes. I like that idea

**Pull request merged**